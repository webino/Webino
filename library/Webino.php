<?php
/**
 * Webino™ (http://webino.sk)
 *
 * @link        https://github.com/webino for the canonical source repository
 * @copyright   Copyright (c) 2015-2018 Webino, s.r.o. (http://webino.sk)
 * @author      Peter Bačinský <peter@bacinsky.sk>
 * @license     BSD-3-Clause
 */

use Webino\Version;
use Webino\App\Application\CoreConfig;
use Webino\App\Factory;
use Webino\App\Options\DebuggerOptions;
use Webino\App\Service\Debugger;
use Webino\App\Service\DebuggerInterface;

/**
 * Class Webino
 */
class Webino
{
    /**
     * Webino™ version
     */
    const VERSION = Version::VERSION;

    /**
     * Create application core config
     *
     * @param array $config
     * @return CoreConfig
     */
    public static function config(array $config)
    {
        return new CoreConfig($config);
    }

    /**
     * Create Webino application
     *
     * @param array|object $config
     * @param DebuggerInterface $debugger
     * @return \WebinoAppLib\Application\AbstractBaseApplication
     */
    public static function application($config = null, DebuggerInterface $debugger = null)
    {
        return (new Factory)->create($config, $debugger);
    }

    /**
     * Create application debugger
     *
     * @param array|DebuggerOptions $options
     * @return Debugger
     */
    public static function debugger($options = [])
    {
        return new Debugger($options);
    }

    /**
     * Create application debugger options
     *
     * @return DebuggerOptions
     */
    public static function debuggerOptions()
    {
        return new DebuggerOptions;
    }
}
