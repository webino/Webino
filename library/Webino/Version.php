<?php
/**
 * Webino™ (http://webino.sk)
 *
 * @link        https://github.com/webino for the canonical source repository
 * @copyright   Copyright (c) 2015-2017 Webino, s.r.o. (http://webino.sk)
 * @author      Peter Bačinský <peter@bacinsky.sk>
 * @license     BSD-3-Clause
 */

namespace Webino;

/**
 * Class to store and retrieve the version of Webino™
 */
final class Version
{
    /**
     * Webino™ version identification
     */
    const VERSION = '0.1.0';
}
