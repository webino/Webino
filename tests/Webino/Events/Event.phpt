<?php
/**
 * Webino™ (http://webino.sk)
 *
 * @link        https://github.com/webino for the canonical source repository
 * @copyright   Copyright (c) 2015-2018 Webino, s.r.o. (http://webino.sk)
 * @author      Peter Bačinský <peter@bacinsky.sk>
 * @license     BSD-3-Clause
 */

use Tester\Assert;
use Webino\Events\EventEmitter;
use Webino\Events\EventInterface;

require __DIR__ . '/../../bootstrap.php';


$emitter = new EventEmitter;
$event = 'test.event';

$callback = function (EventInterface $event) use (&$emitCount) {
    $event['test'] = true;
};


$emitter->on($event, $callback);

$event = $emitter->emit($event);

$emitter->off($callback, $event);

$emitter->emit($event);


Assert::true($event['test']);
Assert::type(EventInterface::class, $event);
Assert::same('test', $event->getValue('unknown', 'test'));
