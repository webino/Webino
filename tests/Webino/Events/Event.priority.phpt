<?php
/**
 * Webino™ (http://webino.sk)
 *
 * @link        https://github.com/webino for the canonical source repository
 * @copyright   Copyright (c) 2015-2018 Webino, s.r.o. (http://webino.sk)
 * @author      Peter Bačinský <peter@bacinsky.sk>
 * @license     BSD-3-Clause
 */

use Tester\Assert;
use Webino\Events\EventEmitter;
use Webino\Events\EventInterface;

require __DIR__ . '/../../bootstrap.php';


$emitter = new EventEmitter;
$event = 'test.event';

$emitTest = [];
$callbackA = function (EventInterface $event) use (&$emitTest) {
    return $emitTest[] = 'A';
};

$callbackB = function (EventInterface $event) use (&$emitTest) {
    return $emitTest[] = 'B';
};

$callbackC = function (EventInterface $event) use (&$emitTest) {
    return $emitTest[] = 'C';
};

$callbackD = function (EventInterface $event) use (&$emitTest) {
    return $emitTest[] = 'D';
};


$emitter->on($event, $callbackA, EventInterface::BEGIN);
$emitter->on($event, $callbackB, EventInterface::BEFORE);
$emitter->on($event, $callbackC, EventInterface::AFTER);
$emitter->on($event, $callbackD, EventInterface::FINISH);

$event = $emitter->emit($event);


Assert::same(['A', 'B', 'C', 'D'], $emitTest);
Assert::same(['D', 'C', 'B', 'A'], $event->getResponses()->toArray());
