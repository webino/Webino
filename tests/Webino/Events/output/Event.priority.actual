Webino\Events\EventResponses::__set_state(/* #5160 */ array(
	'responses' => array('D', 'C', 'B', 'A'),
))
