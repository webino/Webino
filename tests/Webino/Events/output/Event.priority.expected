array('D', 'C', 'B', 'A')
